### PR TITLE
persistent changes 2/12/24

### PR DESCRIPTION
**Pull request by agrinmanriv0537**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/shooter/ShooterVortex.java [hunk 1]
```diff
@@ -35,6 +48,10 @@ public void updateInputs(ShooterIOInputsAutoLogged inputs) {
         inputs.bottomShooterSupplyVoltage = topShooterMotor.getBusVoltage();
         inputs.topShooterAmps = topShooterMotor.getOutputCurrent();
         inputs.bottomShooterAmps = topShooterMotor.getOutputCurrent();
-
+        inputs.topShooterPosition = topEncoder.getPosition();
+        inputs.bottomShooterPosition = bottomEncoder.getPosition();
+        inputs.bottomShooterVelocityRotPerMin = bottomEncoder.getVelocity();
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/shooter/ShooterVortex.java:53]: Duplicate

File: src/main/java/frc/robot/subsystems/shooter/ShooterVortex.java [hunk 2]
```diff
@@ -16,7 +17,19 @@ public class ShooterVortex implements ShooterIO {
...
+    /**
+     * Constructor Shooter Subsystem - sets motor and encoder preferences
+     */
+    public ShooterVortex() {
+        topShooterMotor.setIdleMode(IdleMode.kCoast);
+        bottomShooterMotor.setIdleMode(IdleMode.kCoast);
+        topEncoder.setPositionConversionFactor(1.9375);
+        topEncoder.setVelocityConversionFactor(1.9375);
+        bottomEncoder.setPositionConversionFactor(1.9375);
+        bottomEncoder.setVelocityConversionFactor(1.9375);
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/shooter/ShooterVortex.java:29]: Let's change this back to `31.0/16.0`

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 3]
```diff
@@ -20,6 +28,13 @@ public class Shooter extends SubsystemBase {
         new SimpleMotorFeedforward(Constants.ShooterConstants.KS, Constants.ShooterConstants.KV);
     private ShooterIOInputsAutoLogged inputs = new ShooterIOInputsAutoLogged();
 
+    // Mutable holder for unit-safe voltage values, persisted to avoid reallocation.
+    private final MutableMeasure<Voltage> m_appliedVoltage = mutable(Volts.of(0));
+    // Mutable holder for unit-safe linear distance values, persisted to avoid reallocation.
+    private final MutableMeasure<Angle> m_angle = mutable(Rotations.of(0));
+    // Mutable holder for unit-safe linear velocity values, persisted to avoid reallocation.
+    private final MutableMeasure<Velocity<Angle>> m_velocity = mutable(RPM.of(0));
+
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:37]: This can stay in the test branch for now

**Review by legoguy1000 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/shooter/ShooterVortex.java (hunk 1 above)
Comment (agrinmanriv0537) [reply src/main/java/frc/robot/subsystems/shooter/ShooterVortex.java:53]: topShooterPosition duplicate deleted

File: src/main/java/frc/robot/subsystems/shooter/ShooterVortex.java (hunk 2 above)
Comment (agrinmanriv0537) [reply src/main/java/frc/robot/subsystems/shooter/ShooterVortex.java:29]: changed

**Review by aidan-underwood — APPROVED** [review]

**Review by legoguy1000 — APPROVED** [review]

**Pull request merged**